### PR TITLE
Makes Security Bomb Suit Immune to Explosions Again. + Adds New Juggernaut Suit.

### PR DESCRIPTION
**Pull request by Dark1Star3s**
Security's Bomb Suit would die against explosions ever since the fragmention buff, but now I have fixed that by giving it balistics resistence of a tac plate. (As it should have in the first place considering what it is.)

* Buffed: The Sec Bombsuit to now be immune to explosions once more.
---
I have also given antags a new armor suit, the best armor an antag can get that's not space proof. The Juggernaut suit.
It is immune against ballistics bar the very high AP ones such as AMRs, Flechette Rifles etc.
It has very high melee resist.
It is immune to explosions.
It has decent laser resist, however a carbine will tear right through the suit with ease.
It however is much slower, and has slight restricted vision around the edges of one's screen.

* Added: New Sprites for the Juggernaut Suit. https://i.gyazo.com/e61f009e9ac9422a5e7a0c25c996eec4.png https://i.gyazo.com/3959bea13aa6cf45154ee4e009a55ae5.png
* Added: New Armor Kit in the traitor/merc uplink, contains the Juggernaut Suit for 32 TCs.

### CODE REVIEW

File: code/modules/clothing/suits/utility.dm [hunk 1]
```diff
@@ -109,6 +109,47 @@
 	allowed = list(/obj/item/weapon/gun/energy,/obj/item/weapon/melee/baton,/obj/item/weapon/handcuffs)
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|FEET|ARMS|HANDS
 
+/obj/item/clothing/suit/bomb_suit/juggernaut
+	name = "heaivly armored bombsuit"
+	desc = "A bomb suit, which seems to have a lot of armor plating strapped to it- god that's actually a lot... wow. 'You are up against the wall. I am the fucking wall.'"
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:114]: Make the description sane.

**Comment by Sparrowbird611** [comment]
Jarvis, the pay day memes are getting to your head, and getting in the way of your balancing ideas.

Why the hell would I wear that helmet, which gives slowdown and less armor than other helmet types, when I can just steal a better one that won't restrict my vision. Make the stats consistent with the body armor, or don't bother.

File: code/modules/clothing/suits/utility.dm [hunk 2]
```diff
@@ -109,6 +109,47 @@
...
+		melee = ARMOR_MELEE_VERY_HIGH,
+		bullet = ARMOR_BALLISTIC_AP,
+		laser = ARMOR_LASER_MAJOR,
+		energy = ARMOR_ENERGY_RESISTANT,
+		bomb = ARMOR_BOMB_SHIELDED
+		)
+	flags_inv = HIDEJUMPSUIT|HIDETAIL
+	heat_protection = UPPER_TORSO|LOWER_TORSO
+	max_heat_protection_temperature = ARMOR_MAX_HEAT_PROTECTION_TEMPERATURE
+	siemens_coefficient = 0
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:129]: Why is it shockproof on top of all the armor?

File: code/modules/clothing/suits/utility.dm [hunk 3]
```diff
@@ -109,6 +109,47 @@
...
+	armor = list(
+		melee = ARMOR_MELEE_VERY_HIGH,
+		bullet = ARMOR_BALLISTIC_AP,
+		laser = ARMOR_LASER_MAJOR,
+		energy = ARMOR_ENERGY_RESISTANT,
+		bomb = ARMOR_BOMB_SHIELDED
+		)
+	flags_inv = HIDEJUMPSUIT|HIDETAIL
+	heat_protection = UPPER_TORSO|LOWER_TORSO
+	max_heat_protection_temperature = ARMOR_MAX_HEAT_PROTECTION_TEMPERATURE
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:128]: Why is it heatproof?

File: code/modules/clothing/suits/utility.dm [hunk 4]
```diff
@@ -109,6 +109,47 @@
...
+	tint = TINT_MODERATE
+	armor = list(
+		melee = ARMOR_MELEE_VERY_HIGH,
+		bullet = ARMOR_BALLISTIC_RIFLE,
+		laser = ARMOR_LASER_HANDGUNS,
+		energy = ARMOR_ENERGY_RESISTANT,
+		bomb = ARMOR_BOMB_SHIELDED
+		)
+
+/obj/item/clothing/suit/bomb_suit/juggernaut
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:148]: Why are you declaring this again with totally different properties?

File: code/modules/clothing/suits/utility.dm [hunk 5]
```diff
@@ -109,6 +109,47 @@
...
+	flags_inv = HIDEJUMPSUIT|HIDETAIL
+	heat_protection = UPPER_TORSO|LOWER_TORSO
+	max_heat_protection_temperature = ARMOR_MAX_HEAT_PROTECTION_TEMPERATURE
+	siemens_coefficient = 0
+
+/obj/item/clothing/suit/bomb_suit/juggernaut/New()
+	..()
+	slowdown_per_slot[slot_wear_suit] = 3
+
+/obj/item/clothing/head/bomb_hood/juggernaut
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:135]: Lacks a description unless you mean to use the bomb hood's.

File: code/modules/clothing/suits/utility.dm [hunk 6]
```diff
@@ -109,6 +109,47 @@
 	allowed = list(/obj/item/weapon/gun/energy,/obj/item/weapon/melee/baton,/obj/item/weapon/handcuffs)
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|FEET|ARMS|HANDS
 
+/obj/item/clothing/suit/bomb_suit/juggernaut
+	name = "heaivly armored bombsuit"
+	desc = "A bomb suit, which seems to have a lot of armor plating strapped to it- god that's actually a lot... wow. 'You are up against the wall. I am the fucking wall.'"
+	icon_state = "juggernautsuit"
+	w_class = ITEM_SIZE_HUGE
+	gas_transfer_coefficient = 0.01
+	permeability_coefficient = 0.01
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:118]: Just.. why?

Comment (Dark1Star3s) [reply code/modules/clothing/suits/utility.dm:118]: The only difference between this and the normal bombsuit is the armor vaules. Everything else is the same.

File: code/modules/clothing/suits/utility.dm [hunk 7]
```diff
@@ -109,6 +109,38 @@
 	allowed = list(/obj/item/weapon/gun/energy,/obj/item/weapon/melee/baton,/obj/item/weapon/handcuffs)
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|FEET|ARMS|HANDS
 
+/obj/item/clothing/suit/bomb_suit/juggernaut
+	name = "heavily  armored bombsuit"
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:113]:
```suggestion
	name = "heavily armored bombsuit"
```

File: code/modules/clothing/suits/utility.dm [hunk 8]
```diff
@@ -109,6 +109,38 @@
...
+	desc = "A heavily  modified bombsuit, this one seems to have been made in to a walking tank."
```
Comment (Roland410) [inline code/modules/clothing/suits/utility.dm:114]:
```suggestion
	desc = "A heavily modified bombsuit, this one seems to have been made in to a walking tank."
```

**Pull request merged**